runtime/common/src/fees.rs

```diff
@@ -120,16 +122,11 @@ macro_rules! impl_foreign_fees {
         pub(crate) fn get_fee_factor(
             asset_id: CurrencyId,
         ) -> Result<Balance, TransactionValidityError> {
-            let location = AssetConvert::convert(asset_id);
-            let metadata = location
-                .and_then(|loc| {
-                    <AssetRegistry as orml_traits::asset_registry::Inspect>::metadata_by_location(
-                        &loc,
-                    )
-                })
-                .ok_or(TransactionValidityError::Invalid(InvalidTransaction::Custom(
+            let metadata = <AssetRegistry as Inspect>::metadata(&asset_id).ok_or(
```

Comment:
Nice! It really hurts that I see the simpler API call right next to the previously used :D https://github.com/open-web3-stack/open-runtime-module-library/blob/73b4ee04e402f240f270bfe83f59fc2837dbaf51/asset-registry/src/impls.rs#L185-L191 .. I really don't know why I shot myself in the foot.

Reply:
I know right, asked myself the same thing in respect to my review.